### PR TITLE
bug fixed: MYR-348 handled menu option after removing a friend

### PR DESCRIPTION
**Pull request by immhrmwn**

### CODE REVIEW

File: src/components/profile/friend-list.component.tsx [hunk 1]
```diff
@@ -153,9 +161,10 @@ const FriendsList = ({ profile }: Props) => {
     setOpenModal(!openModal);
   };
 
-  const handleUnFriendRequest = (friendStatus: ExtendedFriend) => {
-    cancelFriendRequest(friendStatus);
+  const handleUnFriendRequest = (friend: ExtendedFriend | null) => {
+    if (friend) cancelFriendRequest(friend);
```
Comment (rei-gun) [inline src/components/profile/friend-list.component.tsx:165]: do you mean `cancelUnFriendRequest`? kalau emang friend then then unfriend request should go through

File: src/components/profile/header.component.tsx [hunk 2]
```diff
@@ -78,20 +78,20 @@ export default function Header({ isAnonymous, user, profile, loading, isGuest }:
   };
 
   const handleUnFriendRequest = () => {
-    cancelFriendRequest(friendStatus);
+    if (friendStatus) cancelFriendRequest(friendStatus);
```
Comment (rei-gun) [inline src/components/profile/header.component.tsx:81]: mungkin lebih readable kalau 
```suggestion
    if (isFriend) cancelFriendRequest(friendStatus);
```

**Review by rei-gun — COMMENTED** [review]
what does `FriendStatus` and `ExtendedFriend` mean?

File: src/components/profile/friend-list.component.tsx (hunk 1 above)
Comment (immhrmwn) [reply src/components/profile/friend-list.component.tsx:165]: there is 2 function 'handleUnfriendRequest' & 'handlecancelFriendRequest' yang gunain cancelFriendRequest function.
cancelFriendRequest gunanya untuk delete friendstatus yang sudah ada.
bedanya handleUnfriendRequest: friend statusnya approved kalau, handlecancelFriendRequest: statusnya masih pending.
karena fungsinya sama, jadi cancelFriendRequest dipake dikeduanya. atau mungkin baiknya dipisah?

File: src/components/profile/header.component.tsx (hunk 2 above)
Comment (immhrmwn) [reply src/components/profile/header.component.tsx:81]: but friendStatus is not boolean type, itu isinya:
`
{
"id": "60de3df28b77cd4adaf2fc0a",
"status": "approved",
"createdAt": "2021-07-01T22:13:06.000Z",
"friendId": "0x5cb52a37da79e5698c31da276f326ceed461b90024a1135648606823f4a7ed62",
"requestorId": "0x700978e3b5215dda80d1e3bcdf44a0caa7e5ed1f004f0c9ea7dbf6381b89393e"
},
`

**Review by rei-gun — APPROVED** [review]

**Pull request merged**